### ISSUE TITLE
[REVIEW]: StarBLAST: a scalable BLAST+ solution for the classroom

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#102]
**Submitting author:** @cosimichele (<a href="http://orcid.org/0000-0001-7609-1939">Michele Cosi</a>)
**Repository:** <a href="https://github.com/LyonsLab/StarBLAST/" target ="_blank">https://github.com/LyonsLab/StarBLAST/</a>
**Version:** 1.0.0a
**Editor:** @JasonJWilliamsNY
**Reviewer:** @oneilsh, @AstrobioMike 
**Archive:** <a href="https://doi.org/10.5281/zenodo.4688218" target="_blank">10.5281/zenodo.4688218</a>

## Status

[![status](https://jose.theoj.org/papers/783fb8d5604fb6e9edb74db031d12a83/status.svg)](https://jose.theoj.org/papers/783fb8d5604fb6e9edb74db031d12a83)

Status badge code:

```
HTML: <a href="https://jose.theoj.org/papers/783fb8d5604fb6e9edb74db031d12a83"><img src="https://jose.theoj.org/papers/783fb8d5604fb6e9edb74db031d12a83/status.svg"></a>
Markdown: [![status](https://jose.theoj.org/papers/783fb8d5604fb6e9edb74db031d12a83/status.svg)](https://jose.theoj.org/papers/783fb8d5604fb6e9edb74db031d12a83)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/LyonsLab/StarBLAST/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@oneilsh & @AstrobioMike , please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @JasonJWilliamsNY know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/LyonsLab/StarBLAST/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 1.0.0a
- [x] **Authorship:** Has the submitting author (@cosimichele) made substantial contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation? (and documentation is sufficient?)
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this software and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies? (Ideally these should be handled with an automated package management solution.)
- [x] **Example usage:** Do the authors include examples of how to use the software?
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state the need for this software and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/LyonsLab/StarBLAST/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 1.0.0a
- [x] **Authorship:** Has the submitting author (@cosimichele) made substantial contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation? (and documentation is sufficient?)
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this software and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies? (Ideally these should be handled with an automated package management solution.)
- [x] **Example usage:** Do the authors include examples of how to use the software?
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state the need for this software and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @oneilsh, @AstrobioMike  it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/jose-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/jose-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by JasonJWilliamsNY** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1371/journal.pcbi.1002243 is OK
- 10.1187/09-11-0087 is OK
- 10.1093/nar/gkn201 is OK
- 10.1093/molbev/msz185 is OK
- 10.1186/1471-2105-10-421 is OK
- 10.1145/2443416.2443417 is OK
- 10.12688/f1000research.8935.1 is OK
- 10.1371/journal.pbio.1002342 is OK
- 10.1145/2792745.2792774 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00102/jose.00102/10.21105.jose.00102.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00102/jose.00102/10.21105.jose.00102.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00102/jose.00102/10.21105.jose.00102.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00102/jose.00102/10.21105.jose.00102.pdf) :page_facing_up: :point_left:

**Comment by JasonJWilliamsNY** [comment]
@oneilsh @AstrobioMike this (#102) is the issue for review. You each have a checklist of steps to make reviewing easier and there are PDFs above where you can the short paper and any changes @CosiMichele  makes in response to your comments. What should follow is really a conversation where everyone participates with some suggestions. If you are unclear about anything please let me know and I will try and help. Once your checklists and reviews are complete I can move forward. 

I'll ask our bot Whedon to set an initial reminder for two weeks. Of course, you will have to take the time you need, so as things come up just let everyone know if more time is needed. I think everyone is flexible, realizes the circumstances of our day, and appreciates your contribution as a volunteer. Thanks!

**Comment by JasonJWilliamsNY** [comment]
@whedon remind @AstrobioMike in 2 weeks

**Comment by whedon** [comment]
Reminder set for @AstrobioMike in 2 weeks

**Comment by JasonJWilliamsNY** [comment]
@whedon remind @oneilsh in 2 weeks

**Comment by whedon** [comment]
Reminder set for @oneilsh in 2 weeks

**Comment by AstrobioMike** [comment]
@JasonJWilliamsNY I'm embarrassed to say I missed the 7-day window to accept the collaboration invitiation, and now I can't interact with my checkboxes :( Any chance we can re-send that?

**Comment by AstrobioMike** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by JasonJWilliamsNY** [comment]
> @JasonJWilliamsNY I'm embarrassed to say I missed the 7-day window to accept the collaboration invitiation, and now I can't interact with my checkboxes :( Any chance we can re-send that?

@AstrobioMike Let me see if I can help

**Comment by JasonJWilliamsNY** [comment]
@AstrobioMike I resent and invite. Please let me know if that allows you to complete the review. @oneilsh I see a pending invite for you, so pinging just in case a reminder catches you before the invite expires. 
Thanks!

**Comment by AstrobioMike** [comment]
I'm good now, thanks!

**Comment by AstrobioMike** [comment]
@JasonJWilliamsNY, just noting I have worked closely with Sateesh Peri (one of the co-authors) doing a few workshops within the past 4 years, though we have not published anything together and do not work at the same organization. I do consider us friends, however, I can assure that this wouldn't affect my ability to provide an impartial assessment of this work, so I request that this conflict be waived. Please let me know if this is ok. Sorry I missed his name in the author list earlier.

**Comment by JasonJWilliamsNY** [comment]
> @JasonJWilliamsNY, just noting I have worked closely with Sateesh Peri (one of the co-authors) doing a few workshops within the past 4 years, though we have not published anything together and do not work at the same organization. I do consider us friends, however, I can assure that this wouldn't affect my ability to provide an impartial assessment of this work, so I request that this conflict be waived. Please let me know if this is ok. Sorry I missed his name in the author list earlier. 

Thanks for disclosing @AstrobioMike. As long as you feel you can provide objective review, your disclosure does not appear to meet the COI definition referenced at the top of this issue. You should be fine to proceed but if you have further concerns just let me know.

**Comment by AstrobioMike** [comment]
Nice work, @CosiMichele et al! Overall the purpose, infrastructure, and documentation seem great :) Thank you for your work on this!

**First I'll comment on specific boxes from the JOSE review checklist that I have yet to check:**

- [ ] **Version: Does the release version given match the GitHub release (v1.0.0)?**
  * No, there are no releases on github, but maybe this is because you are waiting to release once the review process is done?

- [ ] **Example usage: Do the authors include examples of how to use the software?**
  * There is documentation for installation, and usage such as adding a custom database, but I didn't see any example usage documentation for actually using the program – though this is/would be minimal as it's a very clean interface, and I understand the assumption is people know how to use the web blast already. But maybe a short page for general usage could be added to the end of the documentation? Something with images/descriptions of the input and out interfaces?

- [ ] **Tests: Are there automated tests or manual steps described.**
  * Just noting, these don't exist, but i checked this box because I don't think this really applies here.

- [ ] **Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support**
  * No explicit guidelines are currently presented for third parties wishing to contribute, report issues or problems, or seek support. Maybe a page could be added to the documentation providing contact info and/or a link to the github issues page stating these things can be done there?

**I made some minor suggestions/clarifications in this pull request: https://github.com/LyonsLab/StarBLAST/pull/29**
As noted there, I don't consider any of it as required from my point of view.

**Other minor snags/suggestions:**

- For adding our own database, following instructions here: https://starblast.readthedocs.io/en/latest/2_StarBLAST-VICE.html#adding-your-own-databases-to-starblast-vice
    - 3 different ones come up currently when searching for "Create BLAST Database"
       - the first one on the list currently, put up by Alice Minotto, seems to not be the one we want (for this one we need to adjust molecule type in parameters to try to run it, and it ultimately fails) 
       - so should specify which should be used better
    - In the instructions to use this database in the same area, it says to select the folder containing our database, but the folder i created for this seems to only hold a directory with the name and date of the "analysis" launched to create the db, and within that a logs directory. I can't seem to see the actual db files. When choosing the directory I created, it still works, but it should be made clear in the docs here that the user shouldn't expect to see the actual files (if that is the norm)

**General notes for the review:**
- I was able to only fully test the StarBLAST-VICE version, due to not having an HPC to utilize for this and not having a current XSEDE allocation open. Maybe @oneilsh will be able to test those, but given my reviewing of the full documentation, my experience with the VICE version, and that those others are getting a little into more technical grounds anyway (particularly the HPC avenue), I don't think it's a big deal if we can't directly independently test them.

**Comment by whedon** [comment]
:wave: @oneilsh, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @AstrobioMike , please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @oneilsh, please update us on how your review is going.

**Comment by labarba** [comment]
Hi @CosiMichele 👋  — Do give us a quick update about your plan for addressing the comments and suggestions contributed by @AstrobioMike. 

I've also sent a ping via email to our second reviewer, to learn about his status. These are tough times for most people, and we do our best.

**Comment by oneilsh** [comment]
Hi @labarba, @CosiMichele, @JasonJWilliamsNY,

My sincere apologies - this got caught up in a post-personal-leave swirl, but I must also have my github notifications misconfigured. Thank you @labarba for reaching out directly. I am still interested in doing the review, do I need to be re-assigned?

**Comment by labarba** [comment]
Thank you @oneilsh ! Your review checklist is at the top: try to check off an item. If you can, all is well. If not, we'll reissue the invitation (which gives you the permissions).

**Comment by oneilsh** [comment]
Thanks @labarba :) I cannot - I think I will need the invitation re-issued, I must have missed the window.

**Comment by labarba** [comment]
@whedon re-invite @oneilsh as reviewer

**Comment by whedon** [comment]
The reviewer already has a pending invite.

@oneilsh please accept the invite by clicking this link: https://github.com/openjournals/jose-reviews/invitations

**Comment by oneilsh** [comment]
Hmm, I was able to accept the new invite, but I'm not able to interact with the checkboxes or see any options for editing the issue comment.

**Comment by labarba** [comment]
@openjournals/dev --- I have reached a dead-end with a reviewer permissions issue. Could you take a look? The reviewer is unable to edit the checklist, despite being re-invited and accepting the invitation.

**Comment by oneilsh** [comment]
Hi @CosiMichele, @JasonJWilliamsNY,

I'll add some comments here while I work on getting checkbox access :) 

First, I think this is an excellent effort - bioinformatics teaching is challenging, particularly given the heavy resource needs for most bioinformatics software, the very bursty nature of teaching those methods, and how difficult such software is to manage. I appreciate the multiple deployment options presented, though I think the DE VICE version is the most exciting, providing an easy on-ramp for those without HPC resources at their disposal. I also appreciate that this project doesn't re-invent wheels, utilizing SequenceServer which is a great piece of software. 

Some comments on boxes that I'm not sure I can check right away:

**Release version: **
As mentioned by the other reviewer, there are no 'releases' listed on GitHub, but I'm not sure that makes sense for the VICE deployment (which I think is managed inside the DE?) I confess I'm not too familiar with GitHub -> Docker release machinery, but you might find https://docs.github.com/en/actions/guides/publishing-docker-images useful for that aspect. Since a "release" of this package might involve updates to more than one docker image (master and worker), a strategy for tagging (with semantic versioning?) those images and tying them to the repo-described version (v1.0.0 here) would be wise I think. 

**Installation: **
I also don't have access to JetStream or an HPC infrastructure with the required dependencies, so I was only able to test the DE VICE version. It worked - I think the screenshots in the JetStream deployment docs help a lot, having something comparable (or a walk-through video) would help with the DE version since CyVerse DE isn't straightforward for those not used to it. I had trouble opening the app for example until I figured out which DE UI element did the trick.

The HPC documentation is technical and terse, but on first blush seems complete enough, assuming a compatible environment is available. That I think is hard to achieve though: I have access to multiple "HPC" systems, but none with the specific set of pre-reqs suggested (modules properly configured, iRODS, cctools, pbs or similar batch-queue, and user-level ability to run docker). If targeting batch clusters is a strong goal, I would suggest looking into what few common stacks are available - OpenOnDemand is one possibility, but beyond a 1.0 release I suspect. The docs in the HPC section suggest iRODS as a requirement, but I see there are non-iRODS docker images available but their usage is glossed over. 

**Example Usage: **
I think examples of usage for instructors in particular would be most helpful. Many would come from using NCBI's GUI and have no experience setting up infrastructure. The SequenceServer docs fortunately provide a good reference for newcomers and could be selectively referenced. References for signing up for CyVerse would also be useful. It looks like with the DE version I can link to results via URL - how long do those links last? Could students safely submit their work by providing those links in e.g. homework submissions? Suggested workflows for incorporating StarBLAST into the classroom would go a long way to reaching the audience I think, especially if they include installation as well as usage walk-throughs with screenshots or video. 

**Community Guidelines: **
Contribution guidance isn't given, but I think this could be addressed easily by suggesting using Issues for issues or questions and possibly a "seeking help" section of the README with suggestions (e.g. training materials leveraging the tools or guides for customization). Some extra documentation for navigating the codebase might be appreciated by those wanting to dive in to customizations or make improvements.

**Comment by CosiMichele** [comment]
Hi everyone, @AstrobioMike , I believe I have addressed all the issues which you have brought us:

1. Although we believe that this is StarBLAST 1.0, we will be awaiting for this review to pass before assigning the version number. However, if required, we will be happy to add the version to our GitHub.
2. We have added a page for usage and examples (https://starblast.readthedocs.io/en/latest/5_SS_usage.html), including images from our StarBLAST-Docker release. Please let us know if this is satisfying.
3. A small section (Contacts & Issues) has been added at the bottom of our rtd homepage, linking our GitHub issues page.
4. I understand that the process to add a Database to StarBLAST-VICE was unclear; I've modified that section to better reflect steps to undertake in order to add a personalized database. Again, if more needs to be added, let us know.

Additionally, I have merged the small edits suggested in the pull request.

Thank you @AstrobioMike for the feedback!
Finally you @labarba and @oneilsh for checking on this. Let us know what else we can do!

**Comment by arfon** [comment]
> @openjournals/dev --- I have reached a dead-end with a reviewer permissions issue. Could you take a look? The reviewer is unable to edit the checklist, despite being re-invited and accepting the invitation.

Could you tell me what browser you're using?

**Comment by AstrobioMike** [comment]
Thanks for addressing everything from me, @CosiMichele, all looks good to me. And thanks again to you and the co-authors for your work on this!

**Comment by CosiMichele** [comment]
Hi everyone, thank you once again for your time and patience.

Thank you @AstrobioMike for taking time and reviewing StarBLAST! Your input has really helped us.

Thank you @oneilsh for your comments! Here are some updates:

1. Since both you and @AstrobioMike pointed out the version release, I went ahead and added it to the GitHub code.
2. I have edited the HPC requirements installation, addressing how to install iRODS (and ncbi-BLAST+, CCTools) for non users. Hopefully this will be helpful.
3. I had added example usage and community guidelines in my previous edit and highlighted in my previous comment - let us know if further changes are required.

Once again, thank you all for your time and feedback. We look forward to hearing from you!

**Comment by oneilsh** [comment]
@arfon - Chrome, 88.0.4324.192 (Official Build) (x86_64)

@CosiMichele nice updates! They do address most of my thoughts. My only remaining suggestion would be to include some screenshots on the installation/access of the DE version.

**Comment by CosiMichele** [comment]
@oneilsh Thank you for the suggestion! I have added screenshots to help users navigate the DE. 

Please let us know if there's anything else we should do/change.

**Comment by oneilsh** [comment]
It works! Thank you :)

**Comment by JasonJWilliamsNY** [comment]
Hi @oneilsh @AstrobioMike @CosiMichele I wanted to check in and see if there is anything I can do to help move things forward. Looks like some progress has been made over the last month!

**Comment by CosiMichele** [comment]
Hi @JasonJWilliamsNY, thanks for checking in!

I have made the changes proposed by @AstrobioMike and @oneilsh to the documentation. I believe there's only one last proposition from @oneilsh, which has been addressed. As things stand, I am just waiting for feedback on these last changes. Thanks again for checking in and all the reviewers' work!

**Comment by JasonJWilliamsNY** [comment]
@CosiMichele Great! I'll keep my eye out for moving this to the next stage!

**Comment by AstrobioMike** [comment]
Hey all, yep all is good from my perspective. @CosiMichele addressed everything from me :) Thanks!

**Comment by oneilsh** [comment]
Hi @JasonJWilliamsNY same here, let me know if there's anything else needed from my end 👍

**Comment by JasonJWilliamsNY** [comment]
@CosiMichele I take @oneilsh and @AstrobioMike 's comments to mean that they are good with changes they have seen. Are there any other changes you are working on, if not I'll have a pass to get this moving through the next steps.

**Comment by CosiMichele** [comment]
@JasonJWilliamsNY currently we are not working on any further changes. Thanks to the reviewers for your time on this!

**Comment by JasonJWilliamsNY** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00102/jose.00102/10.21105.jose.00102.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00102/jose.00102/10.21105.jose.00102.pdf) :page_facing_up: :point_left:

**Comment by JasonJWilliamsNY** [comment]
@CosiMichele I have reviewed the proof and things look good. The next step is for you to create an archive on either https://zenodo.org/ or https://figshare.com/ and obtain a DOI (which you'll need to give to us). I can then move to the final acceptance and publication steps. 

Tip: Here is a nice guide on doing the above with Zenodo: https://guides.github.com/activities/citable-code/#:~:text=Making%20Your%20Code%20Citable%201%20Choose%20your%20repository.,a%20shiny%20new%20badge%20showing%20your%20new%20DOI%21

**Comment by CosiMichele** [comment]
@JasonJWilliamsNY thank you so much!

I generated the DOI using Zenodo: here's a link https://zenodo.org/badge/latestdoi/223219222  and the badge itself [![DOI](https://zenodo.org/badge/223219222.svg)](https://zenodo.org/badge/latestdoi/223219222)

Let us know what else we can do!

**Comment by labarba** [comment]
Hi @CosiMichele 👋 
I see we're getting to the end with this one 🎉 

We like to ask that the Zenodo metadata be edited to match the JOSE paper: that is match the title, and list of authors (note that the auto-sync with Zenodo lists anyone who has made a commit as author.) Could you make that change?

**Comment by labarba** [comment]
@whedon set 10.5281/zenodo.4688218 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4688218" target="_blank">10.5281/zenodo.4688218</a> is the archive.

**Comment by labarba** [comment]
Sorry for butting in, @JasonJWilliamsNY — I wanted to have a go at the `whedon` superpowers again!

**Comment by labarba** [comment]
Should the version be marked as `1.0.0a`?

**Comment by CosiMichele** [comment]
@labarba , thanks for checking in! I believe I changed the Zenodo metadata to match the JOSE metadata (including the version). Can you please check and let us know if there's more to chance? Thank you!

**Comment by labarba** [comment]
@whedon set 1.0.0a as version

**Comment by whedon** [comment]
OK. 1.0.0a is the version.

**Comment by JasonJWilliamsNY** [comment]
@labarba can you confirm everything is set on your end? I'll assume you'll close the issue at publication?

**Comment by labarba** [comment]
Hi everybody! 👋 
Re-emerging here after a cycle of giving all the spinning plates a little push.
I see that both Reviewers approved the submission: thank you for your contributions, @oneilsh, @AstrobioMike 🙏 
So this is ready to be published. Let's do the magic.

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1371/journal.pcbi.1002243 is OK
- 10.1187/09-11-0087 is OK
- 10.1093/nar/gkn201 is OK
- 10.1093/molbev/msz185 is OK
- 10.1186/1471-2105-10-421 is OK
- 10.1145/2443416.2443417 is OK
- 10.12688/f1000research.8935.1 is OK
- 10.1371/journal.pbio.1002342 is OK
- 10.1145/2792745.2792774 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/jose-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/jose-papers/pull/49

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/jose-papers/pull/49, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSE!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/jose-papers/pull/50
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/jose.00102](https://doi.org/10.21105/jose.00102)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @CosiMichele, your JOSE paper is published 🚀 

Huge thanks to the Editor: @JasonJWilliamsNY and Reviewers: @oneilsh, @AstrobioMike for their contribution—this experiment in alternative publishing is possible thanks to you!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://jose.theoj.org/papers/10.21105/jose.00102/status.svg)](https://doi.org/10.21105/jose.00102)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jose.00102">
  <img src="https://jose.theoj.org/papers/10.21105/jose.00102/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://jose.theoj.org/papers/10.21105/jose.00102/status.svg
   :target: https://doi.org/10.21105/jose.00102
```
This is how it will look in your documentation:

[![DOI](https://jose.theoj.org/papers/10.21105/jose.00102/status.svg)](https://doi.org/10.21105/jose.00102)

**We need your help!**

Journal of Open Source Education is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. 
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-jose

**Comment by JasonJWilliamsNY** [comment]
@labarba can you have a look - ready to publish?

**Comment by labarba** [comment]
hi @JasonJWilliamsNY – did you post on the wrong thread, perhaps? This is already published a while back…

**Comment by JasonJWilliamsNY** [comment]
Doing this from my phone - maybe the app placed me in the wrong thread!

**Comment by JasonJWilliamsNY** [comment]
@labarba check #179

**Comment by labarba** [comment]
:loudspeaker:  New Reviewer Sign-up Form!
https://forms.gle/Rizd3TcHnQKhrbYY7
